File: 프로그래머스/lv3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>
using namespace std;
bool visited[201]; //방문검사

void dfs(vector<vector<int>> &arr, int idx){
    for(int i=0; i<arr[idx].size(); i++){
        if(arr[idx][i] == 1 && visited[i] == false){ //1이고 아직 방문하지 않은 컴
            arr[idx][i] = 0;
            visited[i] = true;
            dfs(arr, i); //dfs 진행
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    for(int i=0; i<n; i++){
        if (visited[i]==false){
            dfs(computers,i); //dfs 진행
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def dfs(n, computers, visited, v):
    print("dfs 호출" ,v)
    visited[v] = True
    for i in range(n):
        if computers[v][i] == 1 and v!=i: # 컴퓨터가 연결되어있고 v가 자기 자신이 아니면
            if not visited[i]:
                dfs(n, computers, visited, i)

def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]
    for i in range(n):
        if not visited[i]:
            dfs(n, computers, visited, i)
            answer += 1
    return answer